File: Going Dutch/MCPayment.m
//
//  MCPayment.m
//  We all pay
//
//  Created by Mark Cornelisse on 23/07/14.
//  Copyright (c) 2014 Mark Cornelisse. All rights reserved.
//

#import "MCPayment.h"
#import "MCCurrency.h"
#import "MCExchangeRate+CoreDataProperties.h"
#import "MCPaymentPresence.h"
#import "MCPerson.h"
#import "MCSharedBill.h"
#import "MCCurrency+CoreDataProperties.h"
#import "MCPayment+CoreDataProperties.h"


@implementation MCPayment

#pragma mark - NSManagedObject

- (void)awakeFromInsert {
    [super awakeFromInsert];
    
    [self setPrimitiveValue:NSUUID.UUID.UUIDString forKey:@"uniquePaymentId"];
    NSDate *now = NSDate.date;
    [self setPrimitiveValue:now forKey:@"dateCreated"];
    [self setPrimitiveValue:now forKey:@"dateModified"];
}

#pragma mark - NSObject

@end

File: Going Dutch/MCPayment.swift
//
//  MCPayment.swift
//  We all pay
//
//  Created by Mark Cornelisse on 26/08/2025.
//  Copyright © 2025 Mark Cornelisse. All rights reserved.
//

import UIKit

@objc(MCPayment) public final class MCPayment: NSManagedObject {

    // MARK: NSManagedObject
    
    public override func awakeFromInsert() {
        super.awakeFromInsert()
        
        self.setPrimitiveValue(UUID().uuidString, forKey: #keyPath(MCPayment.uniquePaymentId))
        let now = Date()
        self.setPrimitiveValue(now, forKey: #keyPath(MCPayment.dateCreated))
        self.setPrimitiveValue(now, forKey: #keyPath(MCPayment.dateModified))
    }
    
    // MARK: NSObject
}
